src/main/groovy/org/shipkit/internal/gradle/plugin/PluginValidationPlugin.java

```diff
@@ -0,0 +1,36 @@
+package org.shipkit.internal.gradle.plugin;
+
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.shipkit.internal.gradle.util.TaskMaker;
+
+/**
+ * This plugin validates that every plugin has a corresponding .properties file.
+ *
+ * Adds tasks:
+ * <ul>
+ *     <li>'validatePlugins' - of type {@link PluginValidatorTask}.
+ *          Validates that every plugin has a corresponding .properties file.</li>
+ * </ul>
+ */
+public class PluginValidationPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(final Project project) {
+        project.getPlugins().withId("java", new Action<Plugin>() {
+            @Override
+            public void execute(Plugin plugin) {
+                TaskMaker.task(project, "validatePlugins", PluginValidatorTask.class, new Action<PluginValidatorTask>() {
+                    @Override
+                    public void execute(PluginValidatorTask task) {
+                        task.setDescription("Validates Gradle Plugins and their properties files");
+                        task.setGradlePlugins(PluginUtil.discoverGradlePlugins(project));
```

Comment:
It's good to push all the logic that is potentially expensive (File IO) and error prone (complexity) to Gradle's execution phase. This way, most invocations of Gradle build are faster and also task can be easily excluded/disabled if it is problematic.

I think we can do all discovering in task action. What do you think?